test/jasmine/tests/shapes_test.js

```diff
@@ -382,6 +382,71 @@ describe('Test shapes:', function() {
     });
 });
 
+describe('shapes axis reference changes', function() {
+    'use strict';
+
+    var gd;
+
+    beforeEach(function(done) {
+        gd = createGraphDiv();
+
+        Plotly.plot(gd, [
+            {y: [1, 2, 3]},
+            {y: [1, 2, 3], yaxis: 'y2'}
+        ], {
+            yaxis: {domain: [0, 0.4]},
+            yaxis2: {domain: [0.6, 1]},
+            shapes: [{
+                xref: 'x', yref: 'paper', type: 'rect',
+                x0: 0.8, x1: 1.2, y0: 0, y1: 1,
+                fillcolor: '#eee', layer: 'below'
+            }]
+        }).then(done);
+    });
+
+    afterEach(destroyGraphDiv);
+
+    function getShape(index) {
+        var s = d3.selectAll('path[data-index="' + index + '"]');
+        expect(s.size()).toBe(1);
```

Comment:
before the fix above, the shape was drawn twice in its initial configuration `xref: 'x', yref: 'paper'`: once on subplot `xy` and again on `xy2`